static_src/components/app_page.jsx

```diff
@@ -48,12 +53,15 @@ export default class AppPage extends React.Component {
 
   render() {
     let content = <h4 className="test-none_message">No app</h4>;
```

Comment:
So just to think through, I'd guess this is something like this:

```
let content = <h4>{ (this.state.loading) ? 'Loading...' : 'No app' }</h4>

if (!this.state.loading && this.state.app.name) { // not loading and app has data
  content = 
  ...
```

Not sure if that's valid react now that I look at it. But the point would be to always show loading state if loading and show no data if app data is missing.